Last-Update: 2015-04-02
Forwarded: https://github.com/ZoneMinder/ZoneMinder/pull/776
Bug-Debian: http://bugs.debian.org/619813
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Description: Use libv4l1-videodev.h from libv4l-dev (>= 0.8.3-1), as it is no
             longer shipped as linux/videodev.h in linux-libc-dev.

--- a/configure.ac
+++ b/configure.ac
@@ -353,9 +353,9 @@
 AC_CHECK_HEADERS(execinfo.h,,,)
 AC_CHECK_HEADERS(ucontext.h,,,)
 AC_CHECK_HEADERS(sys/syscall.h,,,)
 AC_CHECK_HEADERS(pthread.h,,,)
-AC_CHECK_HEADERS(linux/videodev.h,AC_SUBST(ZM_HAS_V4L1,1),AC_SUBST(ZM_HAS_V4L1,0),)
+AC_CHECK_HEADERS(libv4l1-videodev.h,AC_SUBST(ZM_HAS_V4L1,1),AC_SUBST(ZM_HAS_V4L1,0),)
 AC_CHECK_HEADERS(linux/videodev2.h,AC_SUBST(ZM_HAS_V4L2,1),AC_SUBST(ZM_HAS_V4L2,0),)
 if test "$ZM_HAS_V4L1" == "1" || test "$ZM_HAS_V4L2" == "1"; then
 AC_SUBST(ZM_HAS_V4L,1)
 else
--- a/src/zm_local_camera.h
+++ b/src/zm_local_camera.h
@@ -25,11 +25,11 @@
 #include "zm_image.h"
 
 #if ZM_HAS_V4L
 
-#ifdef HAVE_LINUX_VIDEODEV_H
-#include <linux/videodev.h>
-#endif // HAVE_LINUX_VIDEODEV_H
+#ifdef HAVE_LIBV4L1_VIDEODEV_H
+#include <libv4l1-videodev.h>
+#endif // HAVE_LIB4VL1_VIDEODEV_H
 #ifdef HAVE_LINUX_VIDEODEV2_H
 #include <linux/videodev2.h>
 #endif // HAVE_LINUX_VIDEODEV2_H
 
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -95,9 +95,9 @@
 # This is required to enable searching in lib64 (if exists), do not change
 set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS ON)
 
 # System checks
-check_include_file("linux/videodev.h" HAVE_LINUX_VIDEODEV_H)
+check_include_file("libv4l1-videodev.h" HAVE_LIBV4L1_VIDEODEV_H)
 check_include_file("linux/videodev2.h" HAVE_LINUX_VIDEODEV2_H)
 check_include_file("execinfo.h" HAVE_EXECINFO_H)
 check_include_file("ucontext.h" HAVE_UCONTEXT_H)
 check_include_file("sys/sendfile.h" HAVE_SYS_SENDFILE_H)
@@ -396,19 +396,19 @@
 # Setting to zeros first is required because ZM uses #define for these
 set(ZM_HAS_V4L 0)
 set(ZM_HAS_V4L1 0)
 set(ZM_HAS_V4L2 0)
-if(HAVE_LINUX_VIDEODEV_H)
+if(HAVE_LIBV4L1_VIDEODEV_H)
 	set(ZM_HAS_V4L 1)
 	set(ZM_HAS_V4L1 1)
-endif(HAVE_LINUX_VIDEODEV_H)
+endif(HAVE_LIBV4L1_VIDEODEV_H)
 if(HAVE_LINUX_VIDEODEV2_H)
 	set(ZM_HAS_V4L 1)
 	set(ZM_HAS_V4L2 1)
 endif(HAVE_LINUX_VIDEODEV2_H)
-if((NOT HAVE_LINUX_VIDEODEV_H) AND (NOT HAVE_LINUX_VIDEODEV2_H))
+if((NOT HAVE_LIBV4L1_VIDEODEV_H) AND (NOT HAVE_LINUX_VIDEODEV2_H))
 	message(AUTHOR_WARNING " Video 4 Linux headers weren't found - Analog and USB camera support will not be available")
-endif((NOT HAVE_LINUX_VIDEODEV_H) AND (NOT HAVE_LINUX_VIDEODEV2_H))
+endif((NOT HAVE_LIBV4L1_VIDEODEV_H) AND (NOT HAVE_LINUX_VIDEODEV2_H))
 # Check for PCRE and enable ZM_PCRE accordingly
 set(ZM_PCRE 0)
 if(HAVE_LIBPCRE AND HAVE_PCRE_H)
 	set(ZM_PCRE 1)
--- a/zoneminder-config.cmake
+++ b/zoneminder-config.cmake
@@ -3,9 +3,8 @@
 
 /* This file is used by cmake to create config.h for ZM */
 
 /* General system checks */
-#cmakedefine HAVE_LINUX_VIDEODEV_H 1
 #cmakedefine HAVE_LINUX_VIDEODEV2_H 1
 #cmakedefine HAVE_EXECINFO_H 1
 #cmakedefine HAVE_UCONTEXT_H 1
 #cmakedefine HAVE_SYS_SENDFILE_H 1
@@ -51,8 +50,9 @@
 #cmakedefine HAVE_LIBSWSCALE 1
 #cmakedefine HAVE_LIBSWSCALE_SWSCALE_H 1
 #cmakedefine HAVE_LIBVLC 1
 #cmakedefine HAVE_VLC_VLC_H 1
+#cmakedefine HAVE_LIBV4L1_VIDEODEV_H 1
 
 /* Authenication checks */
 #cmakedefine HAVE_MD5_OPENSSL 1
 #cmakedefine HAVE_MD5_GNUTLS 1
